persistence/remote/src/main/java/org/infinispan/persistence/remote/upgrade/HotRodTargetMigrator.java

```diff
@@ -112,4 +113,16 @@ public void disconnectSource(Cache<Object, Object> cache) throws CacheException
       });
       CompletableFuture.allOf(future).join();
    }
+
+   @Override
+   public void connectSource(Cache<Object, Object> cache, PersistenceConfiguration configuration) {
```

Comment:
I think this should return `CompletionStage<Void>` and do the join in the caller as necesary. Then in the rest server we don't need to use a blocking thread and can use this stage. The JMX method can just block.

Reply:
I didn't worry too much about this method since it a once in a lifetime of the server call, but sure, it'd be better to avoid blocking, I will change it